File: 剑指offer/剑指 Offer 39. 数组中出现次数超过一半的数字.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=1;
        int result=nums[0];
        for(int i=1;i<nums.size();i++){
            if(n==0){
                result=nums[i];
                n=1;

            }
            else if(result==nums[i]) n++;
            else n--;
        }
        return result;

    }
};

File: 剑指offer_java/剑指 Offer 39. 数组中出现次数超过一半的数字.java
class Solution {
    public int majorityElement(int[] nums) {
        int x = 0, votes = 0;
        for(var num:nums){
            if(num==x) votes++;
            else {
                votes--;
                if(votes<=0) {
                    x=num;
                    votes=1;
                }
            }
        }
        return x;
    }
}
